ibis/backends/tests/test_generic.py

```diff
@@ -2027,6 +2024,123 @@ def test_sample_with_seed(backend):
     backend.assert_frame_equal(df1, df2)
 
 
+@pytest.mark.parametrize(
+    "inp, exp",
+    [
+        pytest.param(
+            lambda: ibis.literal(1).cases([(1, "one"), (2, "two")], "other"),
+            "one",
+            id="basic",
+        ),
+        pytest.param(
+            lambda: ibis.literal(1).cases([(1, "one"), (2, "two")], default="other"),
+            "one",
+            id="one_kwarg",
+        ),
+        pytest.param(
+            lambda: ibis.literal(1).cases(
+                case_result_pairs=[(1, "one"), (2, "two")], default="other"
+            ),
+            "one",
+            id="two_kwargs",
+        ),
+        pytest.param(
+            lambda: ibis.literal(1).cases(
+                default="other", case_result_pairs=[(1, "one"), (2, "two")]
+            ),
+            "one",
+            id="two_kwargs_swapped",
+        ),
+        pytest.param(
+            lambda: ibis.literal(5).cases([(1, "one"), (2, "two")], "other"),
+            "other",
+            id="other",
+        ),
+        pytest.param(
+            lambda: ibis.literal(5).cases([(1, "one"), (2, "two")]),
+            None,
+            id="fallthrough",
+        ),
+    ],
+)
+def test_value_cases_deprecated(con, inp, exp):
+    with pytest.warns(FutureWarning):
+        i = inp()
+    result = con.execute(i)
+    if exp is None:
+        assert pd.isna(result)
+    else:
+        assert result == exp
+
+
+@pytest.mark.parametrize(
+    "inp, exp",
+    [
+        pytest.param(
+            lambda: ibis.literal(1).cases((1, "one"), (2, "two"), else_="other"),
+            "one",
+            id="one_kwarg",
+        ),
+        pytest.param(
+            lambda: ibis.literal(5).cases((1, "one"), (2, "two")),
+            None,
+            id="fallthrough",
+        ),
+    ],
+)
+def test_value_cases_scalar(con, inp, exp):
+    result = con.execute(inp())
+    if exp is None:
+        assert pd.isna(result)
+    else:
+        assert result == exp
+
+
+@pytest.mark.broken(
+    "exasol",
+    reason="the int64 RBI column is .to_pandas()ed to an object column, which is incomparable to ints",
+    raises=AssertionError,
+)
+def test_value_cases_column(batting):
+    df = batting.to_pandas()
+    expr = batting.RBI.cases(
+        (5, "five"), (4, "four"), (3, "three"), else_="could be good?"
+    )
+    result = expr.execute()
+    expected = pd.Series(
```

Comment:
Do you need this constructor here? You're passing a sequence to `Counter`, and that should be agnostic to the sequence type.